security-framework/src/os/macos/identity.rs

```diff
@@ -25,11 +25,15 @@ impl SecIdentityExt for SecIdentity {
         keychains: &[SecKeychain],
         certificate: &SecCertificate,
     ) -> Result<Self> {
-        let keychains = CFArray::from_CFTypes(keychains);
+        let keychains = if !keychains.is_empty() {
+            CFArray::from_CFTypes(keychains).as_CFTypeRef()
```

Comment:
does this CFArray live long enough? it looks like it could potentially be dropped at the end of the if block

Reply:
Good point. This is ugly but should be correct now.